CommonData/column.cpp

```diff
@@ -1503,372 +974,416 @@ strintmap Column::labelsSyncStrings(const stringvec &new_values, const strstrmap
+	bool useFilter = filter.size() == rowCount();
+	
+	for(size_t row=0; row<rowCount(); row++)
+		if(!useFilter || filter[row])
+			doubles.push_back(!isEmptyValue(_dbls[row]) ? _dbls[row] : EmptyValues::missingValueDouble);
+				
+	return doubles;
+}
 
-				if(refs == 1) //only this row has this label, so we'll remove it
-				{
-					for(size_t i=0; i<_labels.size(); i++)
-						if(_labels[i] == oldLabel)
-						{
-							_labels.erase(_labels.begin() + i);
-							oldLabel->dbDelete();
-							break;
-						}
-
-					//Make sure users dont change their columns from scalar to nominal text all the time by typing a letter by accident somewhere, so try to convert back if it seems reasonable to do so
-					if(_type != _preEditType && _preEditType == columnType::scale)
-					{
-						double dbl;
-                        if(     (
-                                    !oldLabel ||
-									!convertValueToDoubleForImport(oldLabel->label(),	dbl)
-                                )
-								&&
-								convertValueToDoubleForImport(value,				dbl))
-							typeChanged = changeType(_preEditType) == columnTypeChangeResult::changed; //Just try it
-					}
-					else if(_type == columnType::nominalText && (_preEditType == columnType::nominal || _preEditType == columnType::ordinal))
-					{
-						bool tryValueIntegers = false;
-						int integer;
-						if(     (
-									!oldLabel ||
-									!convertValueToIntForImport(oldLabel->label(),	integer)
-								)
-								&&
-								convertValueToIntForImport(value,				integer))
-						{
-							typeChanged = changeType(_preEditType) == columnTypeChangeResult::changed; //Just try it
-							tryValueIntegers = !typeChanged;
-						}
-						else
-							tryValueIntegers = true;
-
-						if(tryValueIntegers)
-						{
-							//maybe there are non-integer labels yet integer values?
-							bool onlyIntegers = true;
-							int integer;
-							for(size_t i=0; onlyIntegers && i<_labels.size(); i++)
-								if  ( ! (	_labels[i]->originalValue().isInt() ||
-											(_labels[i]->originalValue().isDouble() && double(int(_labels[i]->originalValue().asDouble())) == _labels[i]->originalValue().asDouble())
-										)
-									)
-									onlyIntegers = false;
-
-							if (onlyIntegers)
-								_changeColumnToNominalOrOrdinal(_preEditType);
-						}
-					}
-
-					delete oldLabel;
-
-					_resetLabelValueMap();
-					_dbUpdateLabelOrder();
-				}
-			}
+Label *Column::replaceDoubleWithLabel(double dbl)
+{
+	return replaceDoubleWithLabel(doublevec(dbl))[dbl];
+}
 
-			break;
+std::map<double, Label*> Column::replaceDoubleWithLabel(doublevec dbls)
+{
+	JASPTIMER_SCOPE(Column::replaceDoubleWithLabel);
+	
+	std::map<double, Label*	>	doubleLabelMap;
+	std::map<double, int	>	doubleIntIdMap;
+	
+	beginBatchedLabelsDB();
+	for(double dbl : dbls)
+		if(!doubleIntIdMap.count(dbl))
+		{
+			assert(!std::isnan(dbl)); //Because why would it be?
```

Comment:
How can you be sure that the label does not exist yet?

Reply:
We are in the function "replaceDoubleWithLabel", where we pass the dbls onwards from another function.
However, it makes no sense to replace "nan" in dbls with a label, because that is the value that would be written there if the originalValue is not convertible to double... 

So that wouldnt fit the data at all.
To enforce this and show any problems as soon as possible without confusing behaviour (given that `std::nan != std::nan` im not sure if it could ever be: `doubleIntIdMap.count(std::nan) > 0`)